File: Basics/BST/BST/Class/BinarySearchTree.cs
using System;
using System.Collections.Generic;
using System.Reflection.PortableExecutable;
using System.Text;

namespace BST.Class
{
    public class BinarySearchTree
    {
        public class Node
        {
            public int Value { get; set; }
            public Node Left { get; set; }
            public Node Right { get; set; }

            public Node(int val)
            {
                Value = val;
                Left = null; 
                Right = null;
            }
        }

        public Node Root { get; set; }

        public Node Insert(Node node, int value)
        {
            if(Root == null)
            {
                node = new Node(value);
                Root = node;
                return node;
            }

            else
            {
                if(node == null)
                {
                    node = new Node(value);
                    return node;
                }

                if (value <= node.Value)
                    node.Left = Insert(node.Left, value);

                if (value > node.Value)
                    node.Right = Insert(node.Right, value);
            }
            return node;
        }

        public bool Contains(Node node, int searchValue)
        {
            if (node == null)
                return false;

            else
            {
                if (node.Value == searchValue)
                    return true;

                if (node.Value < searchValue)
                    return Contains(node.Right, searchValue);
                else
                    return Contains(node.Left, searchValue);
            }
        }
    }
}

File: py/binaryTree/binarySearchTree.py
from node import Node
from binaryTree import BinaryTree

class BinarySearchTree:
    def __init__(self):
        self.root = None

    def insert(self, roots: Node, val):
        if(self.root is None):
            node = Node(val)
            self.root = node
        else:
            if(roots is None):
                node = Node(val)
                roots = node
                return roots
            else:
                if(roots.value >= val):
                    roots.left = self.insert(roots.left, val)
                else:
                    roots.right = self.insert(roots.right, val)
                return roots
        
    
    def contains(self, roots, searchVal):
        if(roots is None):
            return False
        else:
            if(roots.value == searchVal):
                return True
            elif(roots.value >= searchVal):
                return self.contains(roots.left, searchVal)
            else:
                return self.contains(roots.right, searchVal)


bst = BinarySearchTree()
bst.insert(bst.root, 5)
bst.insert(bst.root, 3)
bst.insert(bst.root, 4)
bst.insert(bst.root, 7)
bst.insert(bst.root, 8)

result = bst.contains(bst.root, 7)
print(result)
# bt = BinaryTree()
# #bt.preOrder(bst.root)
# #bt.inOrder(bst.root)
# # bt.postOrder(bst.root)
# bt.breadthFirst(bst.root)